clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -252,4 +257,18 @@ public UUIDSerde() {
     static public Serde<byte[]> ByteArray() {
         return new ByteArraySerde();
     }
+
+    static public final class ListSerde<T> extends WrapperSerde<List<T>> {
```

Comment:
Should we change to `ListSerde<L extends List<T>, T> extends WrapperSerde<List<T>>` (cf. `ListDeserializer` comments)

Note: We should not use `WrappedSerde<L>` because the wrapper `Serializer` uses `List<T>` but not `L` (not sure if we want to tie the serializer type to a fixed list type, but I guess we should not).

Reply:
I'm trying to implement `List` type as generic, everything was going well until `WrapperSerde<List<T>>` part. Compiler complains a lot if I try to "generify" `List` as `L` in `Serdes.java`. `ListDe/Serializer` classes are fine.
The best I could do in `Serdes.java`:
```
static public final class ListSerde<L extends List<T>, T> extends WrapperSerde<List<T>> {

    public ListSerde() {
        super(new ListSerializer<>(), new ListDeserializer<>());
    }

    @SuppressWarnings(value = "unchecked")
    public ListSerde(Class<? extends List> listClass, Serde<T> serde) {
        super(new ListSerializer<>(serde.serializer()), new ListDeserializer<>(listClass, serde.deserializer()));
    }

}
```
But then `L` is never used. 

Take a look at my last commit:
https://github.com/yeralin/kafka/commit/de09d01ffdc0f23807ddc9752b62c6244cd5b618